scripts/presubmit.sh

```diff
@@ -1,13 +1,19 @@
 #!/bin/bash
 set -ex
 
+regenerate() {
+    go generate -run="protoc" ./...
+    go generate -run="mockgen" ./...
+}
+trap regenerate EXIT
+
 find . -name "*.pb.go" -delete
 find . -name "mock_*" -delete
 golint --set_exit_status ./...
 go vet ./...
 misspell . -locale US
-go generate -run="protoc" ./...
-go generate -run="mockgen" ./...
+regenerate
```

Comment:
Thanks for making this script more robust. 
I wonder, however, what the appropriate error handling should be.  In the case of an error - eg, protos did not compile successfully, this script should be re-run locally until it passes. In this case, is there a benefit from the added complexity?